### PR TITLE
Resource discovery: Implement channel card

### PR DESCRIPTION
**Pull request by AllanOXDi**
## Summary

…

## References

…

## Reviewer guidance

…

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    :to="to"
+    layout="horizontal"
+    thumbnailDisplay="small"
+    thumbnailAlign="right"
+    :title="title"
+    :headingLevel="headingLevel"
+  >
+    <template #thumbnailPlaceholder>
+      <div>
+        <ChannelThumbnail
```
Comment (MisRob) [inline kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue]: `ChannelThumbnail` component contains plenty of sizing, ratio, styles, and background logic that `KCard` (and internally `KImg` used within it) now takes care of and we don't want to override from consumers. Let's just utilize `KCard` APIs such as `thumbnailSrc` etc. that were made to take care of this. I think I mentioned this in one of the other PRs where I elaborated on reasons for this more. I can't find it now though, so if that's just my bad memory and clarification is needed, let me know and we can chat in more detail. 

You can just put `ContentIcon` into `#thumbnailPlaceholder` and the rest of the things will be taken care of by `KCard`. If the reason it didn't work was https://github.com/learningequality/kolibri-design-system/pull/752, after you resolve the feedback I left there,  `KCard` should support all you need here.

File: kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        default: 2,
+      },
+      contentNode: {
+        type: Object,
+        required: true,
+      },
+    },
+    computed: {
+      numberOfResources() {
+        return `${this.contentNode.num_coach_contents} ${this.$tr('resources')}`;
```
Comment (MisRob) [inline kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue]: We will need to use string pluralization to achieve correct translations (0 resources, 1 resource, ...) https://kolibri-dev.readthedocs.io/en/develop/i18n.html#icu-message-syntax

File: kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      },
+    },
+    $trs: {
```
Comment (MisRob) [inline kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue]: @marcellamaki would you provide guidance on where to place this string? I believe we're already moving away from in-component `$trs` but am not clear myself on where to put this one in. Thank you.

**Comment by MisRob** [comment]
@AllanOXDi please let me know when this is ready for final review. I was just going through all card work so I left some notes here already for important changes. However, I think after that's done, there won't be much else here, perhaps some minor details. Nice work overall.

File: kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue (hunk 3 above)
Comment (marcellamaki) [reply kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue]: It looks like we have a `resourcesLabel` : 'Resources' in `commonCoreStrings` so let's update this to use that string. (Also saves us a translation!) 

For other strings we might run into for the cards, @AlexVelezLl  was just assigned [this issue, which will create a shared file in `kolibri-common` for additional strings](https://github.com/learningequality/kolibri/issues/12609) that may help for anything that will be used for searching/filtering/resources in shared components for this project. Once that's merged, we can place any strings there.

Comment (MisRob) [reply kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleChannelCard.vue]: Oh I see, thanks! Wasn't aware of that.

**Comment by AllanOXDi** [comment]
@MisRob  would you mind giving this another look?

**Comment by MisRob** [comment]
Hi @AllanOXDi, yes - would you please clean up temporary code? As I've already seen this before, I think I will only do a final pre-merge review when everything's ready. Thanks for preparing everything for easier testing :)

**Comment by AllanOXDi** [comment]
Done!

**Comment by AlexVelezLl** [comment]
Hey @AllanOXDi! I just merged #12717 which had some KCard breaking changes. Could you please rebase the branch, and update the `layout` prop to `orientation` as this is one of the breaking changes introduced in the latest KDS version https://github.com/learningequality/kolibri-design-system/releases/tag/v5.0.0-rc7 :)

File: packages/kolibri-common/components/Cards/AccessibleChannelCard.vue [hunk 4]
```diff
@@ -0,0 +1,96 @@
+<template>
+
+  <KCard
+    :to="to"
+    :layout="windowBreakpoint === 0 ? 'vertical' : 'horizontal'"
```
Comment (MisRob) [inline packages/kolibri-common/components/Cards/AccessibleChannelCard.vue]: https://github.com/learningequality/kolibri/pull/12607#issuecomment-2411383764

File: packages/kolibri-common/components/Cards/AccessibleChannelCard.vue [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        required: true,
+      },
+      thumbnailSrc: {
+        type: String,
+        default: null,
+      },
+    },
+    computed: {
+      numberOfResources() {
+        return `${this.contentNode.num_coach_contents} ${this.coreString('resourcesLabel')}`;
```
Comment (MisRob) [inline packages/kolibri-common/components/Cards/AccessibleChannelCard.vue]: https://github.com/learningequality/kolibri/pull/12607#discussion_r1733993053

**Comment by MisRob** [comment]
Two last things, @AllanOXDi, and then can be merged :)

**Review by MisRob — APPROVED** [review]
Thank you @AllanOXDi

**Pull request merged**